src/test/groovy/com/causecode/fileuploader/UtilitiesServiceSpec.groovy

```diff
@@ -0,0 +1,65 @@
+package com.causecode.fileuploader
+
+import grails.test.mixin.TestFor
+import org.springframework.web.multipart.MultipartFile
+import spock.lang.Specification
+
+/**
+ * Unit tests for {@link UtilitiesService}
+ */
+@TestFor(UtilitiesService)
+class UtilitiesServiceSpec extends Specification {
+
+    void "test getProviderInstance method"() {
+        when: 'getProviderInstance method is called and class does not exist'
+        service.getProviderInstance('test')
+
+        then: 'Method should throw exception'
+        ProviderNotFoundException e = thrown()
+        e.message == 'Provider test not found.'
+
+        when: 'getProviderInstance method is called and class exist'
+        service.getProviderInstance('Amazon')
+
+        then: 'No exception is thrown'
+        noExceptionThrown()
+    }
+
+    void "test getNewTemporaryDirectoryPath method"() {
+        when:
+        service.newTemporaryDirectoryPath
+        then:
+        noExceptionThrown()
+    }
+
+    void "test getTempFilePathForMultipartFile"() {
+        when:
+        service.getTempFilePathForMultipartFile('Foo', '.pdf')
+        then:
+        noExceptionThrown()
+    }
+
+    @SuppressWarnings('JavaIoPackageAccess')
+    void "test moveFile"() {
+        setup: 'Dummy Dir'
+        File dir = new File('./dummyDir')
+        dir.mkdir()
+        dir.deleteOnExit()
+
+        when: 'instance is of simpleFile'
+        File f = new File('./test')
+        f.deleteOnExit()
+        service.moveFile(f, dir.absolutePath)
+
+        then:
+        noExceptionThrown()
```

Comment:
Can you check if the file exists at the new destination?

Reply:
Done.